verifier/src/channel.rs

```diff
@@ -407,11 +433,18 @@ impl<E: FieldElement> TraceOodFrame<E> {
             let mut current_aux = vec![E::ZERO; self.aux_trace_width];
             let mut next_aux = vec![E::ZERO; self.aux_trace_width];
 
-            for (i, a) in self.values.chunks(2).skip(self.main_trace_width).enumerate() {
+            for (i, a) in
+                self.main_and_aux_evaluations.chunks(2).skip(self.main_trace_width).enumerate()
+            {
                 current_aux[i] = a[0];
                 next_aux[i] = a[1];
             }
             Some(EvaluationFrame::from_rows(current_aux, next_aux))
         }
     }
+
+    /// Returns the Lagrange kernel evaluation frame, if any.
+    pub fn lagrange_kernel_column_frame(&self) -> Option<&LagrangeKernelEvaluationFrame<E>> {
```

Comment:
minor nit: `lagrange_kernel_frame()` is probably sufficient (i.e., no need for "column").